tests/components/forecast_solar/test_config_flow.py

```diff
@@ -57,6 +57,39 @@ async def test_user_flow(hass: HomeAssistant) -> None:
     assert len(mock_setup_entry.mock_calls) == 1
 
 
+async def test_options_flow_invalid_api(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test options config flow when API key is invalid."""
+    mock_config_entry.add_to_hass(hass)
+    with patch(
+        "homeassistant.components.forecast_solar.async_setup_entry", return_value=True
+    ):
+        await hass.config_entries.async_setup(mock_config_entry.entry_id)
+        await hass.async_block_till_done()
+
+    result = await hass.config_entries.options.async_init(mock_config_entry.entry_id)
+
+    assert result.get("type") == FlowResultType.FORM
+    assert result.get("step_id") == "init"
+    assert "flow_id" in result
+
+    result2 = await hass.config_entries.options.async_configure(
```

Comment:
I'd patch out the integration setup function until the options flow is done and we've awaited `hass.async_block_till_done` to make sure that the config entry reload task has fired.

Reply:
You mean something like this?

```py
with patch(
        "homeassistant.components.forecast_solar.async_setup_entry", return_value=True
    ):
        result = await hass.config_entries.options.async_init(mock_config_entry.entry_id)
        result2 = await hass.config_entries.options.async_configure(
            result["flow_id"],
            user_input={
                CONF_API_KEY: "solarPOWER!",
                CONF_DECLINATION: 21,
                CONF_AZIMUTH: 22,
                CONF_MODULES_POWER: 2122,
                CONF_DAMPING: 0.25,
                CONF_INVERTER_SIZE: 2000,
            },
        )
        await hass.config_entries.async_setup(mock_config_entry.entry_id)
        await hass.async_block_till_done()
```

Reply:
Pushed new changes